### ISSUE TITLE
Proposal: Add some animations to Expander

### ISSUE THREAD

**Issue by ShankarBUS** [Kinnara/ModernWpf#135]
## Summary
This proposal is to add some animations to the **Expander** control as in **WCT**

## Rationale
The only parts this library missing from WinUI are the nice little animations in it.
While animating other controls might be hard, animating the **Expander** control is relatively easy.
I know this due to limitation in WPF and this is how it works. **"But it is not impossible."**

See this codeproject article : [Customizing WPF Expander with ControlTemplate](https://www.codeproject.com/Articles/248112/Templating-WPF-Expander-Control) by [XiaoChuan Yu](https://www.codeproject.com/Members/XiaoChuan-Yu). A basic implementation like this would be appreciated

## Screenshots
**Expander from WCT**
![Expander_WCT](https://user-images.githubusercontent.com/63704247/87247049-ed9a9900-c46e-11ea-810a-b41fbde27e14.gif)

**Expander from this library**
![Expander_MW](https://user-images.githubusercontent.com/63704247/87247124-71ed1c00-c46f-11ea-9fb7-7240a406b292.gif)

## Additional Context
I may be a little biased since I need this for one of my apps. But no would say they don't need fluid animations.
Implementing this will be a nice addition to this library.
If you are okay I will work on this and submit a PR for this

**Comment by Kinnara** [comment]
Good idea. Is there a "standard" expand/collapse animation in UWP/WinUI that we can borrow, to make our animation as native as possible?

**Comment by ShankarBUS** [comment]
There is no expander in WinUI AFAIK, it's from the Windows Community Toolkit (aka WCT), the presets could be there.

**Comment by ShankarBUS** [comment]
Should the header be rotated when in Left/Right Expand mode?

**Comment by ShankarBUS** [comment]
Hey @Kinnara,
Are you planning on implementing this, should I open a PR? Are you gonna implement it yourself?

**Comment by Kinnara** [comment]
If [this proposal](https://github.com/microsoft/microsoft-ui-xaml/issues/3279) is accepted, we can port the control (or just the animations) over.

**Comment by ShankarBUS** [comment]
That's why I closed the related PR.

I don't have a solid decision on whether to port the whole API or just the animations.

Since most of the apps already use the WPF expander, creating a custom style would be a good option. But that shouldn't stop innovation.

IMO, We can go with porting the whole control.

**Comment by ShankarBUS** [comment]
@Kinnara, the proposal was accepted, a PR was made and got mergerd 🎉🥳

https://github.com/microsoft/microsoft-ui-xaml/commit/ce84da3986fde706f4dc02d26851a4598d11e0e0

What about now?

**Comment by Kinnara** [comment]
We probably want to first update the styling of the WPF Expander to match the WinUI one. I'll need to take a closer look to see what's possible and what's not.

**Comment by Kinnara** [comment]
Closing as completed during the 1.x backlog review. The requested behavior is now implemented or superseded by the 1.0.0-preview.1 reboot (#688), or the resolution is already documented in this thread. If a focused problem still reproduces on 1.x, please file a new report against that version.
